drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -294,6 +335,51 @@ local function temp_attr_handler(driver, device, ib, response)
   end
 end
 
+local mired_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value ~= nil then
+      if (ib.data.value < COLOR_TEMPERATURE_MIRED_MIN or ib.data.value > COLOR_TEMPERATURE_MIRED_MAX) then
+        device.log.warn_with({hub_logs = true}, string.format("Device reported a min or max color temperature %d mired outside of supported capability range", ib.data.value))
+        return
+      end
+      local temp_in_kelvin = mired_to_kelvin(ib.data.value)
+      set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..minOrMax, ib.endpoint_id, temp_in_kelvin, false)
+      local min = get_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, ib.endpoint_id)
+      local max = get_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, ib.endpoint_id)
+      if min ~= nil and max ~= nil then
+        if min < max then
+          device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({ value = {minimum = min, maximum = max} }))
+          set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, ib.endpoint_id, nil, false)
+          set_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, ib.endpoint_id, nil, false)
```

Comment:
I'm assuming this is to accommodate for the case where the bounds are changing. However, an event won't be emitted unless both of the bounds change. So, I think rather than unsetting the fields, you can perform a check to see if the field changed in addition to the nil checks.

Reply:
The reasoning for setting these to nil was just to save some memory so these values are no longer stored on the device object once they are not longer needed. As it stands right now, the device will only have its bound checked once and is not set up to handle dynamic bounds changing. I wouldn't expect that to be a common use case though.